nautobot/extras/nautobot_app/__init__.py

```diff
@@ -0,0 +1,136 @@
+from django.apps import AppConfig
+from operator import getitem
+from collections import OrderedDict
+
+from nautobot.extras.plugins.utils import import_object
+from nautobot.extras.registry import registry
+from nautobot.utilities.choices import ButtonColorChoices
+
+
+class NautobotConfig(AppConfig):
+    menu_tabs = "navigation.menu_tabs"
+
+    def ready(self):
+        if self.__module__.endswith(".apps"):
+            self.__module__ = self.__module__[:-5]
+        menu_items = import_object(f"{self.__module__}.{self.menu_tabs}")
+        if menu_items is not None:
+            register_menu_items(menu_items)
+
+
+def register_menu_items(class_list):
+    for nav_tab in class_list:
+        registry["nav_menu"]["tabs"][nav_tab.name] = {
+            "weight": nav_tab.weight,
+            "groups": registry["nav_menu"]["tabs"][nav_tab.name]["groups"]
+            if registry["nav_menu"]["tabs"].get(nav_tab.name)
+            else {},
+            "permissions": [],
+        }
+
+        tab_perms = []
+        for group in nav_tab.groups:
+            registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name] = {
+                "weight": group.weight,
+                "items": registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name]["items"]
+                if registry["nav_menu"]["tabs"][nav_tab.name]["groups"].get(group.name)
+                else {},
+                "permissions": [],
+            }
+
+            group_perms = []
+            for item in group.items:
+                registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name]["items"][item.link] = item
+                group_perms += item.permissions
+            registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name]["permissions"] = group_perms
+            tab_perms += group_perms
```

Comment:
Please use variables as needed to avoid repeated 6-to-8-layer-deep dict lookups and assignments! :-)

```suggestion
        registry_groups = registry["nav_menu"]["tabs"][nav_tab.name]["groups"]
        for group in nav_tab.groups:
            registry_groups[group.name] = {
                "weight": group.weight,
                "items": registry_groups.get(group.name, {}).get("items", {}),
                "permissions": [],
            }

            group_perms = []
            for item in group.items:
                registry_groups[group.name]["items"][item.link] = item
                group_perms += item.permissions
            registry_groups[group.name]["permissions"] = group_perms
            tab_perms += group_perms
```